hypnotoad/gui/gui.py

```diff
@@ -170,17 +182,22 @@ def save_options_as(self):
 
     def update_options_form(self):
         """Update the widget values in the options form, based on the current
-        values in the options dict
+        values in the options object
 
         """
 
         self.options_form.setSortingEnabled(False)
         self.options_form.cellChanged.disconnect(self.options_form_changed)
-        self.options_form.setRowCount(len(self.options) + 1)
+        self.options_form.setRowCount(len(self.options))
 
-        for row, (key, value) in enumerate(sorted(self.options.items())):
+        filtered_options = dict(self.options)
+        # Skip special keys and options handled specially elsewhere
+        del filtered_options["_magic"]
```

Comment:
This needs to be moved above L191, and `len(filtered_options)` used there, otherwise we blank rows for all the removed options